.github/workflows/run-dep-tests.yml

```diff
@@ -0,0 +1,27 @@
+name: Run dependency tests
+
+on:
+  push:
+  pull_request:
+    branches-ignore: [ master ]
+
+jobs:
+  build:
+
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        python-version: ["3.8", "3.9", "3.10", "3.11", "3.12"]
+        os: [ubuntu-latest, macOS-latest, windows-latest]
+
+    steps:
+    - uses: actions/checkout@v4
```

Comment:
nit; can we pin these to specific hashes so we're not flagging on OSSF. Here's an example: https://github.com/boto/boto3/blob/develop/.github/workflows/run-crt-test.yml#L21

Reply:
I can update that. I'll do it before merging.